solr/core/src/java/org/apache/solr/handler/admin/api/CreateCollectionSnapshotAPI.java

```diff
@@ -0,0 +1,171 @@
+          @DefaultValue("false")
+          @QueryParam("followAliases")
+          boolean followAliases,
+      @QueryParam("asyncId") String asyncId)
+      throws Exception {
+
+    final CreateSnapshotResponse response = instantiateJerseyResponse(CreateSnapshotResponse.class);
+    final CoreContainer coreContainer = fetchAndValidateZooKeeperAwareCoreContainer();
+    recordCollectionForLogAndTracing(collName, solrQueryRequest);
+
+    final String collectionName =
+        followAliases
+            ? coreContainer
+                .getZkController()
+                .getZkStateReader()
+                .getAliases()
+                .resolveSimpleAlias(collName)
+            : collName;
+
+    final ClusterState clusterState = coreContainer.getZkController().getClusterState();
+    if (!clusterState.hasCollection(collectionName)) {
+      throw new SolrException(
+          SolrException.ErrorCode.BAD_REQUEST,
+          "Collection '" + collectionName + "' does not exist, no action taken.");
+    }
+
+    final SolrZkClient client = coreContainer.getZkController().getZkClient();
+    if (SolrSnapshotManager.snapshotExists(client, collectionName, snapshotName)) {
+      throw new SolrException(
+          SolrException.ErrorCode.BAD_REQUEST,
+          "Snapshot with name '"
+              + snapshotName
+              + "' already exists for collection '"
+              + collectionName
+              + "', no action taken.");
+    }
+
+    final ZkNodeProps remoteMessage =
+        createRemoteMessage(collName, followAliases, snapshotName, asyncId);
+    final SolrResponse remoteResponse =
+        CollectionsHandler.submitCollectionApiCommand(
+            coreContainer,
+            coreContainer.getDistributedCollectionCommandRunner(),
+            remoteMessage,
+            CollectionParams.CollectionAction.CREATESNAPSHOT,
+            DEFAULT_COLLECTION_OP_TIMEOUT);
+
+    if (remoteResponse.getException() != null) {
+      throw remoteResponse.getException();
+    }
+
+    response.collection = collName;
+    response.followAliases = followAliases;
+    response.snapshotName = snapshotName;
+    response.requestId = asyncId;
+
+    return response;
+  }
+
+  /**
+   * The Response for {@link CreateCollectionSnapshotAPI}'s {@link #createSnapshot(String, String,
+   * boolean, String)}
+   */
+  public static class CreateSnapshotResponse extends SolrJerseyResponse {
+    @Schema(description = "The name of the collection.")
+    @JsonProperty(COLLECTION_PROP)
+    String collection;
+
+    @Schema(description = "The name of the snapshot to be created.")
+    @JsonProperty("snapshot")
+    String snapshotName;
+
+    @Schema(description = "A flag that treats the collName parameter as a collection alias.")
+    @JsonProperty("followAliases")
+    boolean followAliases;
+
+    @JsonProperty("requestId")
+    String requestId;
+  }
+
+  private ZkNodeProps createRemoteMessage(
```

Comment:
[0] There's definitely a tradeoff in raising the visibility of this method, but I've found it useful to make these "createRemoteMessage" implementations `public static` so that it can be more easily unit tested.